src/sketchup-stl/exporter.rb

```diff
@@ -310,15 +326,23 @@ def self.do_options
         # Export and Cancel Buttons
         #
         btn_export = SKUI::Button.new('Export') { |control|
-
           write_setting('export_units'   , OPTIONS['export_units'])
           write_setting('stl_format'     , OPTIONS['stl_format'])
           write_setting('selection_only' , OPTIONS['selection_only'])
-
-          path = select_export_file()
-
-          export(path, OPTIONS) unless path.nil?
           control.window.close
+          export_entities = get_export_entities()
+          if export_entities
+             path = select_export_file()
+             begin
+                export(path, export_entities, OPTIONS) unless path.nil?
+             rescue => exc
```

Comment:
Style guide says to prefer full names over abbreviations. `exc` would be better expanded to `exception` if that was what you meant. However, default catch is `StandardError` in this case.